### PR TITLE
Improve handling of non-square avatars

### PR DESCRIPTION
**Pull request by rfwatson**
Crop avatars before resizing them. This prevents a user's first experience of Gitea being a nasty aspect ratio on their avatar.

## Before

Screenshot from 2019-05-23 23-03-36

## After

Screenshot from 2019-05-23 23-04-58

## Discussion points

* As [mentioned](https://github.com/go-gitea/gitea/issues/1268#issuecomment-348737761) in #1268 it might be nice to improve the UX flow too, so users have more control over the cropping. I haven't attempted this but I feel like it's worth implementing a simple backend fix first. 

## Questions

* ~CI failing due to the missing `github.com/oliamb/cutter` dependency - how should this be included/packaged? Go newbie here so advice/help welcome.~

### CODE REVIEW

**Comment by zeripath** [comment]
You need to run make vendor and commit

**Comment by rfwatson** [comment]
Committed the change to modules.txt but the build still seems to be failing

**Comment by zeripath** [comment]
You haven't run make vendor and committed

**Review by lafriks — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

File: modules/avatar/avatar.go [hunk 1]
```diff
@@ -5,9 +5,15 @@
 package avatar
 
 import (
+	"bytes"
+	"code.gitea.io/gitea/modules/setting"
```
Comment (lunny) [inline modules/avatar/avatar.go]: import sequence.

File: modules/avatar/avatar_test.go [hunk 2]
```diff
@@ -7,7 +7,9 @@ package avatar
 import (
 	"testing"
 
+	"code.gitea.io/gitea/modules/setting"
```
Comment (lunny) [inline modules/avatar/avatar_test.go:11]: As above.

**Review by lunny — CHANGES_REQUESTED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**